### PR TITLE
Slicer updates preparation

### PR DESCRIPTION
**Pull request by hjmjohnson**
These patches allow building this toolkit with C++14, VTK 9, & ITK 5.3.

Fairly simple changes were made to allow building the external packages:
```
 COMPILE_3D4DImageConversion      ON                                                                                                                                                                                                     
 COMPILE_CONVERTITKFORMATS        ON                                                                                                                                                                                                     
 COMPILE_CROPTOOLS                ON                                                                                                                         
 COMPILE_CombineDWIs              ON               
 COMPILE_DTITensorNormalization   ON                                                                                                                                                                                                     
 COMPILE_DWI_NIFTINRRDCONVERSIO   ON                                                                                                                                                                                                     
 COMPILE_IMAGEMATH                ON                                                                                                                                                                                                     
 COMPILE_IMAGESTAT                ON                                                                                                                                                                                                     
 COMPILE_ITKTransformTools        ON                                                                                                                                                                                                     
 COMPILE_ImageNoise               ON                                                                                                                                                                                                     
 COMPILE_MAURERDISTANCETRANSFOR   ON                                                                                                                                                                                                     
 COMPILE_MULTIATLASSEG            ON                                                                                                                                                                                                     
 COMPILE_NormalizeDWIGradientVe   ON                                                                                                                                                                                                     
 COMPILE_POLYDATAMERGE            ON                                                                                                                                                                                                     
 COMPILE_POLYDATATRANSFORM        ON                                                                                                                                                                                                     
 COMPILE_ReadImageHeader          ON   
 ```         

The following packages have not yet been updated.
```
 COMPILE_PolyDataCompression      OFF                                                                                                                                                                                                    
 COMPILE_CURVECOMPARE             OFF                                                                                                                                                                                                  
 COMPILE_CurveCompare             OFF                                                                                                                                                                                                    
 COMPILE_Unsupported              OFF 
```

### CODE REVIEW

**Comment by hjmjohnson** [comment]
@jcfr I'm preparing some of the historical tools for DTI processing for the next release of Slicer.

**Comment by hjmjohnson** [comment]
@styner FYI:  I'm preparing some of the historical tools I've used from your lab for use with the latest version of Slicer.

Are these tools still worth maintaining as a whole, or are some of the tools outdated?

FYI:  I am also updated https://github.com/NIRALUser/DTIProcessToolkit/pull/50 which will depend on this set of changes.

**Comment by styner** [comment]
Thanks for updating these. We are indeed still using these regularly and though some are outdated and do not need to be maintained (from our end): 3D4DImageConversion, DTITensorNormalization, DWI_NIFTINRRDCONVERSION, ImageNoise, anything in Unsupported and one of the curve comparison (not sure which is right one of the 2). Thanks also for updating the dtiprocesstoolkit

File: CMake/niral_utilitiesExternalProject.cmake [hunk 1]
```diff
@@ -25,7 +25,9 @@ function( niral_utilitiesExternalProject projectname)
 
   foreach(var ${_ep_DEPENDS})
     find_package(${var} REQUIRED)
-    include(${${var}_USE_FILE})
+    if(NOT "${var}" STREQUAL "VTK")
+      include(${${var}_USE_FILE})
+    endif()
```
Comment (jcfr) [inline CMake/niral_utilitiesExternalProject.cmake]:
```suggestion
    if("${var}" STREQUAL "VTK")
      continue()
    endif()
    include(${${var}_USE_FILE})
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -105,18 +135,13 @@ if( BUILD_TESTING )
...
   string(REPLACE "." ";" VERSION_LIST ${ITK_VERSION})
   list(GET VERSION_LIST 0 ITK_VERSION_MAJOR)
   list(GET VERSION_LIST 1 ITK_VERSION_MINOR)
   list(GET VERSION_LIST 2 ITK_VERSION_PATCH)
-
-  if(ITK_VERSION_MAJOR GREATER 4 OR ITK_VERSION_MAJOR EQUAL 4 AND ITK_VERSION_MINOR GREATER 12)
-    message(STATUS "USING compilation flag -std=c++11")
-    set(CMAKE_CXX_FLAGS ${CMAKE_CXX_FAGS} -std=c++11)
-  endif()
 endif()
```
Comment (jcfr) [inline CMakeLists.txt]:
```suggestion
```

This block can be removed as the corresponding variable are set by ITK.

See https://github.com/InsightSoftwareConsortium/ITK/blob/v5.3rc03/CMake/ITKConfig.cmake.in#L25-L27

File: MaurerDistanceTransform/CMakeLists.txt [hunk 3]
```diff
@@ -1,13 +1,11 @@
...
-
 project(MaurerDistanceTransform)
 
 # Find SlicerExecutionModel
 find_package(SlicerExecutionModel REQUIRED)
 include(${SlicerExecutionModel_USE_FILE})
  
 # Find ITK
-find_package(ITK 4.4.0 REQUIRED COMPONENTS
+find_package(ITK 5.3.0 REQUIRED COMPONENTS
```
Comment (jcfr) [inline MaurerDistanceTransform/CMakeLists.txt]:
```suggestion
find_package(ITK 5.3 REQUIRED COMPONENTS
```

For sake of consistency

File: Unsupported/TransformDeformationField/CMakeLists.txt [hunk 4]
```diff
@@ -1,15 +1,12 @@
...
 project(TransformDeformationField)
 
-find_package(ITK REQUIRED)
-include(${ITK_USE_FILE})
-
 if( ${ITK_VERSION_MAJOR} VERSION_GREATER 3 )
   message( FATAL_ERROR "TransformDeformationField needs ITKv3 to compile" )
 endif()
 
+find_package(ITK 3 REQUIRED)
```
Comment (jcfr) [inline Unsupported/TransformDeformationField/CMakeLists.txt:7]: Should this be `5.3`


```suggestion
```
```suggestion
find_package(ITK 5.3 REQUIRED)
```

File: CMakeLists.txt [hunk 5]
```diff
@@ -1,12 +1,42 @@
-project(NIRAL_UTILITIES)
+# ==== Define cmake build policies that affect compilation and linkage default behaviors
```
Comment (jcfr) [inline CMakeLists.txt]:
```suggestion
#-----------------------------------------------------------------------------
# Define CMake build policies that affect compilation and linkage default behaviors
#-----------------------------------------------------------------------------
```

File: CMake/niral_utilitiesExternalProject.cmake (hunk 1 above)
Comment (hjmjohnson) [reply CMake/niral_utilitiesExternalProject.cmake]: Done.

File: CMakeLists.txt (hunk 2 above)
Comment (hjmjohnson) [reply CMakeLists.txt]: Done.

File: CMakeLists.txt (hunk 5 above)
Comment (hjmjohnson) [reply CMakeLists.txt]: Done.

File: MaurerDistanceTransform/CMakeLists.txt (hunk 3 above)
Comment (hjmjohnson) [reply MaurerDistanceTransform/CMakeLists.txt]: Done.

File: Unsupported/TransformDeformationField/CMakeLists.txt (hunk 4 above)
Comment (hjmjohnson) [reply Unsupported/TransformDeformationField/CMakeLists.txt:7]: Done.

**Comment by hjmjohnson** [comment]
@styner No problem.  Great tools need a little love once in a while :).   I have also updated all the other packages that this one depends upon and verified that they build.  After those other packages are merged, the hashes of those packages need to be updated for this repo as well.

**Comment by hjmjohnson** [comment]
@styner  NOTE:  This PR should work for all the internally contained source code.

**Comment by bpaniagua** [comment]
@styner, ok to merge?

**Comment by styner** [comment]
Yes, ok to merge

**Comment by hjmjohnson** [comment]
@bpaniagua A follow-up PR will be needed once the external code repos that this depends upon are also updated.

**Pull request merged**